components/batch-system/src/router.rs

```diff
@@ -346,59 +279,36 @@ where
     /// Close the mailbox of address.
     pub fn close(&self, addr: u64) {
         info!("shutdown mailbox"; "region_id" => addr);
-        unsafe { &mut *self.caches.as_ptr() }.remove(&addr);
-        let mut mailboxes = self.normals.lock().unwrap();
-        if let Some(mb) = mailboxes.map.remove(&addr) {
+        if let Some((_, mb)) = self.normals.remove(&addr) {
             mb.close();
         }
-        mailboxes
-            .alive_cnt
-            .store(mailboxes.map.len(), Ordering::Relaxed);
-    }
-
-    pub fn clear_cache(&self) {
-        unsafe { &mut *self.caches.as_ptr() }.clear();
     }
 
     pub fn state_cnt(&self) -> &Arc<AtomicUsize> {
         &self.state_cnt
     }
 
-    pub fn alive_cnt(&self) -> Arc<AtomicUsize> {
-        self.normals.lock().unwrap().alive_cnt.clone()
+    pub fn alive_cnt(&self) -> usize {
+        self.normals.len()
     }
 
     pub fn trace(&self) -> RouterTrace {
-        let alive = self.normals.lock().unwrap().alive_cnt.clone();
+        let alive = self.alive_cnt();
         let total = self.state_cnt.load(Ordering::Relaxed);
```

Comment:
Dead code? seems it's the only place that uses `state_cnt`.

Reply:
in `apply.rs`, it calls `state_cnt().clone()`